inst/include/dust/random/utils.hpp

```diff
@@ -0,0 +1,74 @@
+#ifndef DUST_RANDOM_UTILS_HPP
+#define DUST_RANDOM_UTILS_HPP
+
+#include <dust/random/numeric.hpp>
+
+namespace dust {
+namespace random {
+
+enum xoshiro_mode {STARSTAR, PLUSPLUS, PLUS};
+
+template <typename T, typename U>
+T int_to_real(U value);
+
+template <>
+inline HOSTDEVICE
+double int_to_real(uint64_t value) {
+  return double(value) / double(utils::uint64_max());
+}
```

Comment:
We will test alternatives in a future PR?

Reply:
exactly. I tried swapping out the float case to use the 32 bit generator and it broke the tests, so wanted to do that separately so we can see the fallout more clearly